guides/advanced_user_guide.md

```diff
@@ -0,0 +1,97 @@
+    - **arguments**: Define the key value pairs for the custom task definition. Below are the mandatory fields
+      - **--apiVersion**: Kubernetes API Version for your custom task CRD.
+      - **--kind**: Kubernetes Kind for your custom task CRD.
+      - **--name**: Kubernetes Resource reference name for your custom task CRD.
+
+      Then, you can add any extra arguments to map as the custom task CRD's `spec.params` fields.
+    - **--file_outputs**: List of Tekton results from the custom task. Since the custom task spec doesn't have Tekton results in static compile time, users need to explicitly define the custom task outputs to avoid mismatching between the custom task I/O.
+
+
+### Custom task with Conditions
+When using custom task with conditions, the compiled workload is slightly different from Argo because we are not wrapping condition blocks as sub-pipeline. The [flip-coin-custom-task](/samples/flip-coin-custom-task) example describes how custom tasks are used with KFP conditions.
+
+The benefits for not having the sub-pipeline in Tekton allow us to represent the condition dependencies that are more align in the DSL. In KFP Argo, the [issue #5422](https://github.com/kubeflow/pipelines/issues/5422) describes how the Argo workflow misrepresent the DSL.
+
+Condition DSL
+```python
+@dsl.pipeline(
+    name='Conditional execution pipeline',
+    description='Shows how to use dsl.Condition().'
+)
+def flipcoin_pipeline():
+    flip = flip_coin_op()
+    with dsl.Condition(flip.output == 'heads') as cond_a:
+        random_num_head = random_num_op(0, 9)
+        random_num_head2 = random_num_op(0, 9).after(random_num_head)
+
+    with dsl.Condition(flip.output == 'tails') as cond_b:
+        random_num_tail = random_num_op(10, 19)
+        random_num_tail2 = random_num_op(10, 19).after(random_num_tail)
+
+    print_output2 = print_op('after random_num_head random_num_tail').after(random_num_head, random_num_tail)
+```
+
+When compiled to Argo vs Tekton:
+
+![condition-dependency](/images/condition-dependency.png)
+
+In this case, the Argo workflow task C will always execution after the condition A,B blocks are done despite task A1 and B1 are skipped. Whereas the Tekton workflow task C only executes if task A1 and B1 are completed and not executes Task C if task A1 and B1 are failed or skipped. So Tekton gives a closer representation of the DSL workflow behavior without using any sub-dag/sub-pipeline.
+
+
+### Custom Task Limitations
+Currently, custom tasks don't support timeout, retries, and any Kubernetes pod spec because these fields are not supported by the [Tekton community](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#limitations) and custom tasks may not need any pod to run their workloads. Therefore, there will not be any log, pod reference, and event for custom tasks in the KFP-Tekton API.
```

Comment:
```suggestion
Currently, custom tasks don't support timeout, retries, or any Kubernetes pod spec because these fields are not supported by the [Tekton community](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#limitations) and custom tasks may not need any pod to run their workloads. Therefore, there will not be any logs, pod references, or events for custom tasks in the KFP-Tekton API.
```